### PR TITLE
Quote cursor name to support uppercase letters (required by postgres)

### LINKED ISSUE #28879: Source-postgres case sensitive column issue.  "updatedat" vs  "users.updatedAt"
### Connector Name

source-postgres

### Connector Version

3.1.0

### What step the error happened?

During the sync

### Revelant information

Postgres 3.1.0 sync is failing with on one cloud user ([job link](https://cloud.airbyte.com/workspaces/c1739b1b-758d-4469-b76b-aca6eff460fc/connections/ec4fb1de-e001-4537-abb9-dd956456c082/job-history))

 ` "stacktrace" : "java.lang.RuntimeException: org.postgresql.util.PSQLException: ERROR: column \"updatedat\" does not exist\n  Hint: Perhaps you meant to reference the column \"users.updatedAt\".\n  Position: 10\n\tat io.airbyte.integrations.source.postgres.PostgresQueryUtils.lambda$getCursorBasedSyncStatusForStreams$4(PostgresQueryUtils.java:178)\n\tat java.base/java.lang.Iterable.forEach(Iterable.java:75)\n\tat io.airbyte.integrations.source.postgres.PostgresQueryUtils.getCursorBasedSyncStatusForStreams(PostgresQueryUtils.java:143)\n\tat io.airbyte.integrations.source.postgres.PostgresSource.getIncrementalIterators(PostgresSource.java:530)\n\tat io.airbyte.integrations.source.postgres.PostgresSource.getIncrementalIterators(PostgresSource.java:127)\n\tat io.airbyte.integrations.source.relationaldb.AbstractDbSource.read(AbstractDbSource.java:161)\n\tat io.airbyte.integrations.base.ssh.SshWrappedSource.read(SshWrappedSource.java:74)\n\tat io.airbyte.integrations.base.spec_modification.SpecModifyingSource.read(SpecModifyingSource.java:49)\n\tat io.airbyte.integrations.base.IntegrationRunner.runInternal(IntegrationRunner.java:139)\n\tat io.airbyte.integrations.base.IntegrationRunner.run(IntegrationRunner.java:99)\n\tat io.airbyte.integrations.source.postgres_strict_encrypt.PostgresSourceStrictEncrypt.main(PostgresSourceStrictEncrypt.java:70)\nCaused by: org.postgresql.util.PSQLException: ERROR: column \"updatedat\" does not exist\n  Hint: Perhaps you meant to reference the column \"users.updatedAt\".\n  Position: 10\n\tat org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2676)\n\tat org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2366)\n\tat org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:356)\n\tat org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:496)\n\tat org.postgresql.jdbc.PgStatement.execute(PgStatement.java:413)\n\tat org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:190)\n\tat org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:134)\n\tat com.zaxxer.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52)\n\tat com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeQuery(HikariProxyPreparedStatement.java)\n\tat io.airbyte.integrations.source.postgres.PostgresQueryUtils.lambda$getCursorBasedSyncStatusForStreams$2(PostgresQueryUtils.java:161)\n\tat io.airbyte.db.jdbc.DefaultJdbcDatabase.bufferedResultSetQuery(DefaultJdbcDatabase.java:55)\n\tat io.airbyte.integrations.source.postgres.PostgresQueryUtils.lambda$getCursorBasedSyncStatusForStreams$4(PostgresQueryUtils.java:161)\n\t... 10 more\n",
`

### Relevant log output

_No response_

### Contribute

- [ ] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by rodireich**
## What
Quote cursor field in getting max value query.
Postgres requires quotes to preserve case sensitivity in queries

closes #28879

### CODE REVIEW

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java [hunk 1]
```diff
@@ -151,13 +151,15 @@ public static Map<AirbyteStreamNameNamespacePair, CursorBasedStatus> getCursorBa
...
         LOGGER.info("Querying max cursor value for {}.{}", namespace, name);
+        cursorInfoOptional.ifPresent(cf -> LOGGER.info("*** {}", cf));
         final String cursorField = cursorInfoOptional.get().getCursorField();
         final String cursorBasedSyncStatusQuery = String.format(MAX_CURSOR_VALUE_QUERY,
             cursorField,
             name,
             cursorField,
             cursorField,
             name);
+        LOGGER.info("*** {}", cursorBasedSyncStatusQuery);
```
Comment (nguyenaiden) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java:162]: Remove if not needed. Same with L154

**Review by nguyenaiden — APPROVED** [review]
Awesome catch. Thanks for this!

**Comment by octavia-squidington-iii** [comment]
##  source-alloydb test report (commit [`520138bcc9`](https://github.com/airbytehq/airbyte/pull/28892/commits/520138bcc9d42c60aaad29b136d2b3e2a0082211)) - ❌

⏲️  Total pipeline duration: 02mn29s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-alloydb/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| QA checks                                                             | ✅        |
| Build connector tar                                                   | ✅        |
| Build source-alloydb docker image for platform linux/x86_64           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rodi_uppercase-cursor/1690850720/520138bcc9d42c60aaad29b136d2b3e2a0082211/source-alloydb/3.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-alloydb test
```

**Comment by octavia-squidington-iii** [comment]
##  source-postgres-strict-encrypt test report (commit [`520138bcc9`](https://github.com/airbytehq/airbyte/pull/28892/commits/520138bcc9d42c60aaad29b136d2b3e2a0082211)) - ✅

⏲️  Total pipeline duration: 02mn33s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rodi_uppercase-cursor/1690850720/520138bcc9d42c60aaad29b136d2b3e2a0082211/source-postgres-strict-encrypt/3.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres-strict-encrypt test
```

**Comment by octavia-squidington-iii** [comment]
##  source-postgres test report (commit [`520138bcc9`](https://github.com/airbytehq/airbyte/pull/28892/commits/520138bcc9d42c60aaad29b136d2b3e2a0082211)) - ✅

⏲️  Total pipeline duration: 02mn33s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rodi_uppercase-cursor/1690850720/520138bcc9d42c60aaad29b136d2b3e2a0082211/source-postgres/3.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres test
```

**Comment by octavia-squidington-iii** [comment]
##  source-alloydb-strict-encrypt test report (commit [`520138bcc9`](https://github.com/airbytehq/airbyte/pull/28892/commits/520138bcc9d42c60aaad29b136d2b3e2a0082211)) - ❌

⏲️  Total pipeline duration: 02mn31s 

| Step                                                                                 | Result   |
|:-------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-alloydb-strict-encrypt/metadata.yaml | ✅        |
| Connector version semver check                                                       | ✅        |
| QA checks                                                                            | ✅        |
| Build connector tar                                                                  | ✅        |
| Build source-alloydb-strict-encrypt docker image for platform linux/x86_64           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rodi_uppercase-cursor/1690850720/520138bcc9d42c60aaad29b136d2b3e2a0082211/source-alloydb-strict-encrypt/3.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-alloydb-strict-encrypt test
```

**Comment by octavia-squidington-iii** [comment]
##  source-alloydb-strict-encrypt test report (commit [`ff5d523184`](https://github.com/airbytehq/airbyte/pull/28892/commits/ff5d52318418940fd8153f91124f99f1890ed41b)) - ✅

⏲️  Total pipeline duration: 02mn31s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rodi_uppercase-cursor/1690851496/ff5d52318418940fd8153f91124f99f1890ed41b/source-alloydb-strict-encrypt/3.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-alloydb-strict-encrypt test
```

**Comment by octavia-squidington-iii** [comment]
##  source-alloydb test report (commit [`ff5d523184`](https://github.com/airbytehq/airbyte/pull/28892/commits/ff5d52318418940fd8153f91124f99f1890ed41b)) - ❌

⏲️  Total pipeline duration: 02mn29s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-alloydb/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| QA checks                                                             | ✅        |
| Build connector tar                                                   | ✅        |
| Build source-alloydb docker image for platform linux/x86_64           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rodi_uppercase-cursor/1690851496/ff5d52318418940fd8153f91124f99f1890ed41b/source-alloydb/3.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-alloydb test
```

**Comment by octavia-squidington-iii** [comment]
##  source-postgres-strict-encrypt test report (commit [`ff5d523184`](https://github.com/airbytehq/airbyte/pull/28892/commits/ff5d52318418940fd8153f91124f99f1890ed41b)) - ❌

⏲️  Total pipeline duration: 02mn27s 

| Step                                                                                  | Result   |
|:--------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-postgres-strict-encrypt/metadata.yaml | ✅        |
| Connector version semver check                                                        | ✅        |
| QA checks                                                                             | ✅        |
| Build connector tar                                                                   | ✅        |
| Build source-postgres-strict-encrypt docker image for platform linux/x86_64           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rodi_uppercase-cursor/1690851496/ff5d52318418940fd8153f91124f99f1890ed41b/source-postgres-strict-encrypt/3.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres-strict-encrypt test
```

**Comment by octavia-squidington-iii** [comment]
##  source-postgres test report (commit [`ff5d523184`](https://github.com/airbytehq/airbyte/pull/28892/commits/ff5d52318418940fd8153f91124f99f1890ed41b)) - ❌

⏲️  Total pipeline duration: 02mn34s 

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-postgres/metadata.yaml | ✅        |
| Connector version semver check                                         | ✅        |
| Connector version increment check                                      | ✅        |
| QA checks                                                              | ✅        |
| Build connector tar                                                    | ✅        |
| Build source-postgres docker image for platform linux/x86_64           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rodi_uppercase-cursor/1690851496/ff5d52318418940fd8153f91124f99f1890ed41b/source-postgres/3.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres test
```

**Comment by octavia-squidington-iii** [comment]
##  source-alloydb-strict-encrypt test report (commit [`74de8f1b62`](https://github.com/airbytehq/airbyte/pull/28892/commits/74de8f1b62c6f9f107016dec2ef3d634f59ec198)) - ❌

⏲️  Total pipeline duration: 6.03s 

| Step                                                                                 | Result   |
|:-------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-alloydb-strict-encrypt/metadata.yaml | ✅        |
| Connector version semver check                                                       | ✅        |
| QA checks                                                                            | ✅        |
| Build connector tar                                                                  | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rodi_uppercase-cursor/1690852272/74de8f1b62c6f9f107016dec2ef3d634f59ec198/source-alloydb-strict-encrypt/3.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-alloydb-strict-encrypt test
```

**Comment by octavia-squidington-iii** [comment]
##  source-alloydb test report (commit [`74de8f1b62`](https://github.com/airbytehq/airbyte/pull/28892/commits/74de8f1b62c6f9f107016dec2ef3d634f59ec198)) - ❌

⏲️  Total pipeline duration: 02mn28s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-alloydb/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| QA checks                                                             | ✅        |
| Build connector tar                                                   | ✅        |
| Build source-alloydb docker image for platform linux/x86_64           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rodi_uppercase-cursor/1690852272/74de8f1b62c6f9f107016dec2ef3d634f59ec198/source-alloydb/3.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-alloydb test
```

**Comment by octavia-squidington-iii** [comment]
##  source-postgres-strict-encrypt test report (commit [`74de8f1b62`](https://github.com/airbytehq/airbyte/pull/28892/commits/74de8f1b62c6f9f107016dec2ef3d634f59ec198)) - ❌

⏲️  Total pipeline duration: 02mn31s 

| Step                                                                                  | Result   |
|:--------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-postgres-strict-encrypt/metadata.yaml | ✅        |
| Connector version semver check                                                        | ✅        |
| QA checks                                                                             | ✅        |
| Build connector tar                                                                   | ✅        |
| Build source-postgres-strict-encrypt docker image for platform linux/x86_64           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rodi_uppercase-cursor/1690852272/74de8f1b62c6f9f107016dec2ef3d634f59ec198/source-postgres-strict-encrypt/3.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres-strict-encrypt test
```

**Comment by octavia-squidington-iii** [comment]
##  source-postgres test report (commit [`74de8f1b62`](https://github.com/airbytehq/airbyte/pull/28892/commits/74de8f1b62c6f9f107016dec2ef3d634f59ec198)) - ❌

⏲️  Total pipeline duration: 02mn43s 

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-postgres/metadata.yaml | ✅        |
| Connector version semver check                                         | ✅        |
| Connector version increment check                                      | ✅        |
| QA checks                                                              | ✅        |
| Build connector tar                                                    | ✅        |
| Build source-postgres docker image for platform linux/x86_64           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rodi_uppercase-cursor/1690852272/74de8f1b62c6f9f107016dec2ef3d634f59ec198/source-postgres/3.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres test
```

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java [hunk 2]
```diff
@@ -73,7 +73,7 @@ public record ResultWithFailed<T>(T result, List<io.airbyte.protocol.models.v0.A
       """;
   public static final String MAX_CURSOR_VALUE_QUERY =
       """
-        SELECT %s FROM %s WHERE %s = (SELECT MAX(%s) FROM %s);
+        SELECT "%s" FROM %s WHERE "%s" = (SELECT MAX("%s") FROM %s);
```
Comment (prateekmukhedkar) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java:76]: verified by looking at `NULL_CURSOR_VALUE_WITH_SCHEMA_QUERY` and `NULL_CURSOR_VALUE_NO_SCHEMA_QUERY` that column names are enclosed in double quotes. LGTM.

**Review by prateekmukhedkar — APPROVED** [review]

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java (hunk 1 above)
Comment (rodireich) [reply airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java]: removed. thanks

**Comment by octavia-approvington** [comment]
Approve this <br/> imagine the commander saying yes

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**